app/src/main/java/org/fossasia/openevent/general/utils/AppLinkUtils.kt

```diff
@@ -0,0 +1,34 @@
+package org.fossasia.openevent.general.utils
+
+import android.net.Uri
+import android.os.Bundle
+import org.fossasia.openevent.general.R
+
+const val EVENT_IDENTIFIER = "eventIdentifier"
+const val VERIFICATION_TOKEN = "verificationToken"
+const val RESET_PASSWORD_TOKEN = "resetPasswordToken"
+private const val SEGMENT_VERIFY = "verify"
+private const val QUERY_TOKEN = "token"
+private const val SEGMENT_RESET_PASSWORD = "reset-password"
+private const val SEGMENT_EVENT = "e"
+
+object AppLinkUtils {
+    fun getDestinationId(uri: Uri): Int? {
+        return when (uri.pathSegments.first()) {
+            SEGMENT_EVENT -> R.id.eventDetailsFragment
+            SEGMENT_VERIFY -> R.id.profileFragment
+            SEGMENT_RESET_PASSWORD -> R.id.eventsFragment
+            else -> null
```

Comment:
Personally I would have liked if these were non Android dependent as robo electroc tests take a long time to run.


You can just split by `/` and see the last group. Pure java, mush easier to test

Reply:
If I split using `/` with strings then what does URI parsing test means, Because it should pass through the `AppLinkUtils` class, and it requires uri. IMO one thing can be done - If I pass a string in `AppLinkUtils` class instead of URI and then convert it into uri and parse.

Reply:
Make ut require a string

Reply:
Can I use `MockitoJUnitRunner` because for getting strings from bundles also require to mock the dependencies?

Reply:
Why are you using Bundles here? Why can't it be done using only Java. See the PR in orga app

Reply:
Updated